addons/website/tests/test_res_users.py

```diff
@@ -64,3 +66,30 @@ def create_user_pou():
         # rollback in retrying().
         with TestCase.assertRaises(self, ValidationError), mute_logger('odoo.sql_db'):
             retrying(create_user_pou, env)
+
+    def _create_user_via_website(self, website, login):
+        # We need a fake request to _signup_create_user.
+        with MockRequest(self.env, website=website):
+            return self.env['res.users'].with_context(website_id=website.id)._signup_create_user({
+                'name': login,
+                'login': login,
+            })
+
+    def test_portal_become_internal(self):
+        # This tests if the website_id is correctly removed when a portal user
+        # becomes an internal user.
+        website = self.env['website'].create({'name': "Awesome Website"})
+        # Permit uninvited signup for portal users.
+        website.auth_signup_uninvited = 'b2c'
+        # Disable cross-website for portal users.
+        website.specific_user_account = True
+        user = self._create_user_via_website(website, 'Portal_User')
+        self.assertEqual(user.website_id, website)
+        self.assertEqual(user._is_portal(), True)
+        self.assertTrue("Portal" in user.groups_id.mapped('name'))
```

Comment:
Avoid doing name comparison, it's fragile, and not how we check group on a user anyway, you could do:
```suggestion
        self.assertTrue(user.has_group('base.group_portal'))
```

But it's already what `_is_protal` check the line above does, so this can just be removed?

Side note, when you check for a `in` presence in a `assert` check, use `assertIn`, it's done for that and display a better and easier to understand error message in case of failure.